### ISSUE TITLE
[accessibility] Interactive markers should have a default accessible name

### ISSUE THREAD

**Issue by Malvoz** [mapbox/mapbox-gl-js#11030]
Interactive markers (e.g. markers that bind to popups) should have a default [accessible name](https://www.w3.org/TR/accname/#dfn-accessible-name), e.g. `aria-label="Marker"` if not specified.

Related WCAG 2.1 Success Criterion: [4.1.2 Name, Role, Value](https://www.w3.org/TR/WCAG21/#name-role-value) (this issue is to address the <q>Name</q> part of the Success Criterion for interactive markers).

<hr>

Furthermore, ideally developers should ensure that each marker has a unique and descriptive name, such that the user can know what the marker is visually highlighting (in the case of markers binding to popups it'd also allow <span title="Assistive technology">AT</span> users to know what it's for before opening the popup). For example, the marker in https://docs.mapbox.com/mapbox-gl-js/example/set-popup/ would ideally be named "Washington Monument" or similar. Perhaps developers can be warned if markers are not uniquely named?

**Comment by mourner** [comment]
Thanks so much for the thorough evaluations and hints @Malvoz! I've put all these issues in the `accessibility` milestone — seems like a lot of low-hanging fruits we could address quickly https://github.com/mapbox/mapbox-gl-js/milestone/48

**Comment by fergcody06** [comment]
We recently updated our Mapbox version to v2.6.0 to resolve a WCAG issue we were seeing where role="button" needed to be added to the markers. That issue is now resolved but it introduced a new WCAG issue where Aria buttons must be accompanied by an accessible name. This issue will solve that. Does anyone know when this is going out?
